tests/infra/libbyz/e2e_test.py

```diff
@@ -125,67 +127,82 @@ def replica_checks(
+                        process_view_change = True
+                    if "is smaller than request counter" in line:
+                        counter_broken_logs += 1
+                    if "Fixed" in line:
+                        fix_logs += 1
+                    if "Reply count" in line:
+                        reply_callback = True
+                    if (
+                        test_client_proxy
+                        and "total requests executed" in line
+                        and "total requests executed 0" not in line
+                    ):
+                        operations_complete = True
+
+            assert replica_ready
+            assert syscall_stats
+            assert fix_logs == counter_broken_logs
+            if test_client_proxy:
+                if f != 0 or node.id == 0:
+                    assert operations_complete
+                if f == 0:
+                    assert reply_callback
+
+        if not with_delays or f == 0:
+            # if view changes not forced or running with f == 0 we shouldn't see any
+            assert not send_view_change
+            assert not process_view_change
+
+        if not send_view_change:
+            logger.info(
+                "***************** NO VIEW CHANGES ISSUED DURING THIS TEST *****************"
+            )
+        if not process_view_change:
+            logger.info(
+                "***************** NO VIEW CHANGES PROCESSED DURING THIS TEST *****************"
+            )
+
+        if with_delays:
+            logger.info("with_delays is set so checking for view changes")
+            assert send_view_change
+            assert process_view_change
+
+    except AssertionError:
+        _, _, tb = sys.exc_info()
+        traceback.print_tb(tb)
+        tb_info = traceback.extract_tb(tb)
+        filename, line, func, text = tb_info[-1]
+        logger.error('Assertion failed \"{}\"'.format(text))
 
-        assert replica_ready
-        assert syscall_stats
-        assert fix_logs == counter_broken_logs
-        if test_client_proxy:
-            if f != 0 or node.id == 0:
-                assert operations_complete
-            if f == 0:
-                assert reply_callback
-
-    if not with_delays or f == 0:
-        # if view changes not forced or running with f == 0 we shouldn't see any
-        assert not send_view_change
-        assert not process_view_change
-
-    if not send_view_change:
-        logger.info(
-            "***************** NO VIEW CHANGES ISSUED DURING THIS TEST *****************"
-        )
-    if not process_view_change:
-        logger.info(
-            "***************** NO VIEW CHANGES PROCESSED DURING THIS TEST *****************"
-        )
-
-    if with_delays:
-        logger.info("with_delays is set so checking for view changes")
-        assert send_view_change
-        assert process_view_change
+        [[logger.info(l) for l in lines] for lines in node_logs]
```

Comment:
Odd to make this a comprehension if you don't use the value, rather than a statement? It does work though.